### ISSUE TITLE
Override couscous.yml configuration from command line

### ISSUE THREAD

**Issue by cyberbit** [CouscousPHP/Couscous#159]
This doesn't have to be super complex or super smart in my case, but with modifications it could be pretty powerful. More on that in a bit.

I have a project using Couscous that has three instances on three different servers. Documentation is generated out of a repository separate from the project. Because I use customized resources grabbed from the current server, I have to generate three different sets of documentation, changing the `baseUrl` on each one. I have them listed in `couscous.yml`, and just comment out the ones I don't need when generating.

Would it be possible to have a command-line syntax to override configuration variables at runtime? Perhaps something like this:

```
couscous generate --config:baseUrl=http://prod.server.com/ --config:title="Server.com PROD"
```

Or, to be more advanced (and insane), you could do the above changes and add a built-in configuration selector to shortcut to preset options. For example, in `couscous.yml`:

```
title:
    dev: Server.com DEV
    prod: Server.com PROD

description: A fantastic website

baseUrl:
    dev: https://dev.server.com
    prod: https://prod.server.com
```

At the command-line (`--instance` could be a more specific form of  `--config` as well, perhaps equivalent to `--config:_instance` or something):

```
couscous generate --instance:dev
```

This would temporarily modify the configuration as if `couscous.yml` looked like this:

```
title: Server.com DEV
description: A fantastic website
baseUrl: https://dev.server.com
```

Admittedly, this is a pipe dream. There are limitations and implications that I recognize with this (selecting instance values in sub-options, restricted keywords, processing modified config file in an additional step, etc.). But even the first suggestion would reduce the amount of manual changes needed to publish correctly for my use case significantly, and feels like it would be easy to medium "hardness."

**Comment by mnapoli** [comment]
The first option would be very simple to use and understand, I like it! Maybe something a bit more standard like this:

```
couscous generate --config baseUrl=http://prod.server.com
```

I think that would be compatible with Symfony's syntax.

**Comment by cyberbit** [comment]
Sounds good to me! If I find time I will certainly try my hand at adding this.

**Comment by cyberbit** [comment]
Alright, I have something that seems to work well. I just need to work on tests. If I post a pull request here, will it track updates as I push them? I have not done many pull requests here, I'm afraid.

**Comment by mnapoli** [comment]
@cyberbit yes absolutely, open a PR and you can keep pushing new commits to the same branch, they will appear in the PR (the diff will be updated).

**Comment by mnapoli** [comment]
Done in #187

**Comment by cyberbit** [comment]
@richoconnell The parsing does not currently support dot properties. It would not be hard to parse out the dot syntax. The particular line is here:

https://github.com/CouscousPHP/Couscous/blob/master/src/Module/Config/Step/OverrideConfigFromCLI.php#L36

And the code context:

```php
foreach ($project->metadata['cliConfig'] as $item) {
    $explosion = explode('=', $item, 2);
    $this->logger->notice('Overriding global config: '.$explosion[0].' = "'.$explosion[1].'"');

    $cliConfig[$explosion[0]] = $explosion[1]; // <<< This line
}
```

**Comment by richoconnell** [comment]
@cyberbit ok, thanks for getting back
